src/main/scala-2.12/kamon/scala/instrumentation/FutureInstrumentation.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2014 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+package kamon.scala.instrumentation
+
+import kamon.trace.{TraceContextAware, Tracer}
+import org.aspectj.lang.ProceedingJoinPoint
+import org.aspectj.lang.annotation._
+
+@Aspect
+class FutureInstrumentation {
```

Comment:
The point cuts are different between 2.11 and 2.12.
The private type ```scala.concurrent.impl.Future.PromiseCompletingRunnable``` looks to have been removed in 2.12 and I was attempting to remove all warnings both compile-time and run-time.

They are however, the same between 2.10 and 2.11.  

Having a separate, scala specific source folder was required, as sbt seems to consider the source specific folder as an additional source folder, not an alternate source folder.